### PR TITLE
Fix R package build on CI.

### PR DESCRIPTION
**Pull request by trivialfis**
@hcho3 Is it possible to trigger a build for the R package on the PR branch instead of the master branch?

### CODE REVIEW

**Comment by hcho3** [comment]
@trivialfis You can comment out the `if` condition in this script:
https://github.com/dmlc/xgboost/blob/a83748eb4521581f9f37fc40ae5651c150d720fc/tests/buildkite/build-rpkg-win64-gpu.ps1#L8-L21

**Comment by hcho3** [comment]
@trivialfis We can always build the R package on PRs to ensure that we don't break the build accidentally.

**Comment by hcho3** [comment]
See my latest commit.

**Comment by trivialfis** [comment]
@hcho3 Thank you for the reference. Will try to fix the Linux build as well.

**Comment by hcho3** [comment]
Yeah, skipping the R package build on PRs was not such a bright idea.

**Comment by trivialfis** [comment]
For the longer term, I think we need to make cmake a system requirement https://github.com/dmlc/xgboost/blob/a83748eb4521581f9f37fc40ae5651c150d720fc/R-package/DESCRIPTION#L70

**Comment by hcho3** [comment]
Pretty sure all issues are fixed. Fingers crossed.

**Review by hcho3 — APPROVED** [review]

File: tests/buildkite/build-rpkg-win64-gpu.ps1 [hunk 1]
```diff
@@ -5,12 +5,12 @@ $ErrorActionPreference = "Stop"
 Write-Host "--- Build XGBoost R package with CUDA"
 
 nvcc --version
-if ( $is_release_branch -eq 1 ) {
-  $arch_flag = "-DGPU_COMPUTE_VER=75"
+$arch_flag = "-DGPU_COMPUTE_VER=75"
```
Comment (hcho3) [inline tests/buildkite/build-rpkg-win64-gpu.ps1:8]: This line is actually not used. In a follow-up PR, I will find ways to only build `sm_75` to speed up builds for pull requests. (For `master`, still need to build all `sm`s.)

**Comment by hcho3** [comment]
The CI is now healthy again. Thanks!

**Comment by trivialfis** [comment]
@hcho3 Thank you for the nice work!

**Pull request merged**